modules/virt-configuring-vm-real-time.adoc

```diff
@@ -0,0 +1,105 @@
+
+You can run real-time workloads on virtual machines (VMs).
+
+.Prerequisites
+* Your cluster is configured to run real-time workloads.
+* You have installed the `virtctl` tool.
+
+.Procedure
+. Edit the `VirtualMachine` manifest to include information about CPU topology, CRI-O annotations, and huge pages:
++
+.Example `VirtualMachine` manifest
+[source,yaml]
+----
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+metadata:
+  name: realtime-vm
+spec:
+  template:
+    metadata:
+      annotations:
+        cpu-load-balancing.crio.io: disable <1>
+        cpu-quota.crio.io: disable <2>
+        irq-load-balancing.crio.io: disable <3>
+    spec:
+      domain:
+        cpu:
+          dedicatedCpuPlacement: true
+          isolateEmulatorThread: true
+          model: host-passthrough
+          numa:
+            guestMappingPassthrough: {}
+          realtime: {}
+          sockets: 1 <4>
+          cores: 4 <5>
+          threads: 1
+        devices:
+          autoattachGraphicsDevice: false
+          autoattachMemBalloon: false
+          autoattachSerialConsole: true
+        ioThreadsPolicy: auto
+        memory:
+          guest: 4Gi
+          hugepages:
+            pageSize: 1Gi <6>
+        terminationGracePeriodSeconds: 0
+# ...
+----
+<1> This annotation specifies that load balancing is disabled for CPUs that are used by the container.
+<2> This annotation specifies that the CPU quota is disabled for CPUs that are used by the container.
+<3> This annotation specifies that Interrupt Request (IRQ) load balancing is disabled for CPUs that are used by the container.
+<4> The number of sockets inside the VM. 
+<5> The number of cores inside the VM. This must be a value greater than or equal to `1`.
+<6> The size of the huge pages. The possible values for x86-64 architecture are 1Gi and 2Mi. In this example, the request is for 4 huge pages of size 1Gi.
+
+. Save and exit the editor.
+. Apply the `VirtualMachine` manifest:
++
+[source,terminal]
+----
+$ oc apply -f <file_name>.yaml
+----
+
+. Configure the guest operating system. The following example shows the configuration steps for {op-system-base} 8 operating system:
+.. Run the following command to connect to the VM console:
++
+[source,terminal]
+----
+$ virtctl console <vm_name>
+----
+
+.. Configure huge pages by using the GRUB boot loader command-line interface. In the following example, 8 1G huge pages are specified.
++
+[source,terminal]
+----
+$ grubby --update-kernel=ALL --args="default_hugepagesz=1GB hugepagesz=1G hugepages=8"
+----
+
+.. To achieve low-latency tuning by using the `cpu-partitioning` profile in the TuneD application, run the following commands:
++
+[source,terminal]
+----
+$ dnf install -y tuned-profiles-cpu-partitioning
+----
++
+[source,terminal]
+----
+$ echo isolated_cores=2-9 > /etc/tuned/cpu-partitioning-variables.conf
+----
+The first two CPUs (0 and 1) are set aside for house keeping tasks and the rest are isolated for the real-time application.
```

Comment:
Just wondering if "housekeeping tasks" is too colloquial. Is there a better term we could use instead here?

Reply:
It is a [term used in computing](https://en.wikipedia.org/wiki/Housekeeping_(computing)) as well. Do you have any other suggestions?

Reply:
Fair enough then!